### PR TITLE
Rename default branch to 'main'

### PR DESCRIPTION
**Pull request by lfdebrux**
We've renamed the default branch to main, following best practice on GitHub [[1]], and we want and expect that users will do the same. This is especially the case for new repositories on GitHub, which will have the name 'main' instead of 'master'.

This PR updates the example deploy script to match.

We also add the new default branch name to the config template. This is because in the [tech-docs-gem](github.com/alphagov/tech-docs-gem) we assume the default branch is named 'master', and changing that would be a breaking change. By changing it here we can immediately reach new users of the tech docs template (without affecting our ability to make the breaking change), and it also keeps the example deploy script and the build output consistent.

[1]: https://github.com/github/renaming

### CODE REVIEW

File: optional/script/deploy [hunk 1]
```diff
@@ -18,12 +18,12 @@ if [[ ! -f Staticfile.auth ]]; then
...
+# Check that we're on main, or there's an environment variable set to override
 BRANCH=$(git rev-parse --abbrev-ref HEAD)
 
 echo "Checking branch..."
-if [[ $BRANCH != "master" ]] && [[ $I_REALISE_I_AM_NOT_ON_MASTER != "y" ]]; then
-  error "Current branch is not master, cowardly refusing to deploy"
+if [[ $BRANCH != "main" ]] && [[ $I_REALISE_I_AM_NOT_ON_MASTER != "y" ]]; then
+  error "Current branch is not main, cowardly refusing to deploy"
   error "To force override, run this again with I_REALISE_I_AM_NOT_ON_MASTER=y"
   exit 1
```
Comment (36degrees) [inline optional/script/deploy:28]: I think the 'override' environment variable also needs updating?

```suggestion
if [[ $BRANCH != "main" ]] && [[ $I_REALISE_I_AM_NOT_ON_MAIN != "y" ]]; then
  error "Current branch is not main, cowardly refusing to deploy"
  error "To force override, run this again with I_REALISE_I_AM_NOT_ON_MAIN=y"
  exit 1
```

**Review by 36degrees — COMMENTED** [review]
Gosh, I feel like this is quite a dated way to deploy to CF, but agree it makes sense to bring this in line 👍🏻 

One comment but generally makes sense.

File: optional/script/deploy (hunk 1 above)
Comment (lfdebrux) [reply optional/script/deploy:28]: Hmmm, good point... I wish I understood though what this is here for or if anyone uses it... this feels like it could be a breaking change, especially the way that it interacts with our expectations for the default branch name in the site configuration :/

Comment (36degrees) [reply optional/script/deploy]: I think the idea was to allow for deploys from local machines (i.e. for those not using Travis etc – at the time GitHub Actions wasn't even a thing, I think) and to have this here as a 'safety rail' but give users the ability to override it if they knew what they were doing.

More generally, this repo is only ever really relevant when [creating a new documentation project](https://tdt-documentation.london.cloudapps.digital/create_project/create_new_project/#create-a-new-documentation-project) which uses this repo as a [Middleman Project Template](https://middlemanapp.com/advanced/project-templates/).

At this point the Thorfile is used to ask the user a bunch of questions, including if they're planning to deploy to PaaS.

If they say yes, we copy the script/deploy folder into their documentation project:

https://github.com/alphagov/tech-docs-template/blob/ff2013c81ea630ca487e45f817c4105a522651d3/Thorfile#L98-L106

After that, when updating their project, we [tell them to update the gem instead](https://tdt-documentation.london.cloudapps.digital/maintain_project/use_latest_template/). (Before the gem existed, we got them to do something else clever with used Project Templates which I can't remember at this point)

Basically this shouldn't affect anyone updating, only new projects, so we should be relatively free to make other changes to this file if we needed to.

(Given PaaS is being decommissioned, we should possibly just simplify and remove anything to do with deployment from this template anyway at some point?)

Comment (lfdebrux) [reply optional/script/deploy]: Ah, thank you so much for the context, that's really helpful.

> (Given PaaS is being decommissioned, we should possibly just simplify and remove anything to do with deployment from this template anyway at some point?)

Yeah, good point :'(

Comment (36degrees) [reply optional/script/deploy]: More background on the move to a gem here, in case it's interesting / useful:
https://github.com/alphagov/tech-docs-template/issues/126

Comment (lfdebrux) [reply optional/script/deploy]: The fact it's only used for new projects makes me feel more comfortable about changing it, I am now leaning towards making the change in tech-docs-gem to use 'main' even if it is breaking, and just tell users to figure it out 😅

**Review by 36degrees — APPROVED** [review]
